torch/csrc/jit/script/compiler.cpp

```diff
@@ -161,7 +161,7 @@ struct Environment {
     if (as_simple_value &&
         !findInThisFrame(name) &&
         findInParentFrame(name) &&
-        getBlockOwningKind() == prim::Loop) {
+        (getBlockOwningKind() == prim::Loop || getBlockOwningKind() == prim::If)) {
```

Comment:
You need to change the conditional at L181 as well, right?

Reply:
Change it according to our discussion. Thanks!